File: app/src/main/java/com/flatstack/android/utils/recycler_view/BaseHolder.java
package com.flatstack.android.utils.recycler_view;

import android.support.v7.widget.RecyclerView;
import android.view.View;

/**
 * Created by ereminilya on 14/12/16.
 */
public abstract class BaseHolder<T> extends RecyclerView.ViewHolder {

    public BaseHolder(View itemView) {
        super(itemView);
    }

    protected abstract void bind(T item);
}

File: app/src/main/java/com/flatstack/android/util/recyclerview/BaseHolder.kt
package com.flatstack.android.util.recyclerview

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import kotlinx.android.extensions.LayoutContainer

abstract class BaseHolder<T>(
    override val containerView: View
) : RecyclerView.ViewHolder(containerView), LayoutContainer {
    internal abstract fun bind(item: T)
}
